json/src/lib.rs

```diff
@@ -1120,6 +1120,196 @@ pub struct GetDescriptorInfoResult {
     pub has_private_keys: bool,
 }
 
+/// Models the request options of "getblocktemplate"
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+pub struct GetBlockTemplateOptions {
+    pub mode: GetBlockTemplateOptionsModes,
+    //// List of client side supported softfork deployment
+    pub rules: Vec<GetBlockTemplateOptionsRules>,
+    /// List of client side supported features
+    pub capabilities: Vec<GetBlockTemplateOptionsCapabilities>,
+}
+
+/// Enum to represent client-side supported features
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum GetBlockTemplateOptionsCapabilities {
+    // No features supported yet. In the future this could be. for example, Proposal and Longpolling
+}
+
+/// Enum to representing specific block rules that the requested template
+/// should support.
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum GetBlockTemplateOptionsRules {
+    /// Inidcates that the requested template should support segwit blocks
```

Comment:
There's a typo. But tbh I'd suggest dropping the docs here. Most fields don't have as the ultimate reference for docs is the Core API docs.